src/test/java/jewellery/inventory/unit/service/SaleServiceTest.java

```diff
@@ -180,14 +192,16 @@ void testReturnProductWillThrowsSaleNotFoundException() {
 
   @Test
   void testReturnProductSuccessfullyWithTwoProducts() {
-    product.setPartOfSale(saleTwoProducts);
+    product.setPartOfSale(sale);
     Product productBeforeReturn = product;
+    productPriceDiscount.setProduct(productBeforeReturn);
+    sale.setProducts(List.of(productPriceDiscount));
```

Comment:
We have this repeating in every test here? This needs to be refactored to be part of the setup. From what I see in the previous state, the products were passed in the sale creation. I don't understand why we moved away from it?